From 6ec079ebe3790361dcc159765b2d9bb2461ed42f Mon Sep 17 00:00:00 2001
From: Dan Kephart <dan.kephart@lairdtech.com>
Date: Mon, 24 Oct 2016 19:48:32 -0400
Subject: [PATCH] laird: cfg80211: allow setting of different regulatory
 database filenames

Add a new config option to allow specifying a different name than db.txt
for the internal wireless regulatory database.  In addition, add a text
file for FCC and ETSI databases.

Depends on CFG80211_INTERNAL_REGDB

Bug: 10334
---
 net/wireless/Kconfig     | 13 +++++++++++++
 net/wireless/Makefile    |  4 +++-
 net/wireless/db_etsi.txt | 12 ++++++++++++
 net/wireless/db_fcc.txt  | 13 +++++++++++++
 4 files changed, 41 insertions(+), 1 deletion(-)
 create mode 100644 net/wireless/db_etsi.txt
 create mode 100644 net/wireless/db_fcc.txt

diff --git a/net/wireless/Kconfig b/net/wireless/Kconfig
index da72ed32f143..7476cf71c2df 100644
--- a/net/wireless/Kconfig
+++ b/net/wireless/Kconfig
@@ -174,6 +174,19 @@ config CFG80211_INTERNAL_REGDB
 
 	  Most distributions have a CRDA package.  So if unsure, say N.
 
+config CFG80211_INTERNAL_REGDB_FILE
+	string "specify file for internal regulatory rules" if CFG80211_INTERNAL_REGDB
+	default "db.txt"
+	depends on CFG80211_INTERNAL_REGDB
+	---help---
+	  This option specifies what file in net/wireless is used to
+	  generate the internal data structure repesenting the wireless
+	  regulatory rules.  Normally, db.txt is used but this allows
+	  the kernel to be preloaded with multiple text files specifying
+	  different regulatory rules.
+
+	  If unsure, say "db.txt".
+
 config CFG80211_CRDA_SUPPORT
 	bool "support CRDA" if CFG80211_INTERNAL_REGDB
 	default y
diff --git a/net/wireless/Makefile b/net/wireless/Makefile
index 4c9e39f04ef8..2e8c2f2cfd7d 100644
--- a/net/wireless/Makefile
+++ b/net/wireless/Makefile
@@ -19,7 +19,9 @@ CFLAGS_trace.o := -I$(src)
 
 ccflags-y += -D__CHECK_ENDIAN__
 
-$(obj)/regdb.c: $(src)/db.txt $(src)/genregdb.awk
+$(eval $(call config_filename,CFG80211_INTERNAL_REGDB_FILE))
+
+$(obj)/regdb.c: $(src)/$(CFG80211_INTERNAL_REGDB_FILE_FILENAME) $(src)/genregdb.awk
 	@$(AWK) -f $(srctree)/$(src)/genregdb.awk < $< > $@
 
 clean-files := regdb.c
diff --git a/net/wireless/db_etsi.txt b/net/wireless/db_etsi.txt
new file mode 100644
index 000000000000..d479d4d21e59
--- /dev/null
+++ b/net/wireless/db_etsi.txt
@@ -0,0 +1,12 @@
+# This is the etsi regulatory domain
+country 00: DFS-ETSI
+	(2400 - 2483.5 @ 40), (100 mW)
+	(5150 - 5250 @ 80), (100 mW), NO-OUTDOOR, AUTO-BW
+	(5250 - 5350 @ 80), (100 mW), NO-OUTDOOR, DFS, AUTO-BW
+	(5470 - 5725 @ 160), (500 mW), DFS
+	# short range devices (ETSI EN 300 440-1)
+	(5725 - 5875 @ 80), (25 mW)
+	# 60 GHz band channels 1-4 (ETSI EN 302 567)
+	(57000 - 66000 @ 2160), (40)
+
+
diff --git a/net/wireless/db_fcc.txt b/net/wireless/db_fcc.txt
new file mode 100644
index 000000000000..eaed166d3531
--- /dev/null
+++ b/net/wireless/db_fcc.txt
@@ -0,0 +1,13 @@
+# This is the fcc regulatory domain
+country 00: DFS-FCC
+	(2402 - 2472 @ 40), (30)
+	# 5.15 ~ 5.25 GHz: 30 dBm for master mode, 23 dBm for clients
+	(5170 - 5250 @ 80), (23), AUTO-BW
+	(5250 - 5330 @ 80), (23), DFS, AUTO-BW
+	(5490 - 5730 @ 160), (23), DFS
+	(5735 - 5835 @ 80), (30)
+	# 60g band
+	# reference: http://cfr.regstoday.com/47cfr15.aspx#47_CFR_15p255
+	# channels 1,2,3, EIRP=40dBm(43dBm peak)
+	(57240 - 63720 @ 2160), (40)
+
-- 
2.11.0

